### PR TITLE
feat(sampling): Add prioritize key transaction option - [TET-547]

### PR DESCRIPTION
**Pull request by priscilawebdev**
Add the option "Prioritize by key transaction" to the dynamic sampling project configuration

**Preview:**

image

ps: backend has to be ready before we can merge

BE PR: https://github.com/getsentry/sentry/pull/41493

### CODE REVIEW

**Review by andriisoldatenko — APPROVED** [review]
pls don't merge until BE is done.

File: static/app/types/sampling.tsx [hunk 1]
```diff
@@ -147,6 +147,7 @@ export type UniformModalsSubmit = (props: {
 export enum DynamicSamplingBiasType {
   BOOST_ENVIRONMENTS = 'boostEnvironments',
   BOOST_LATEST_RELEASES = 'boostLatestRelease',
+  BOOST_KEY_TRANSACTION = 'boostKeyTransaction',
```
Comment (matejminar) [inline static/app/types/sampling.tsx:150]: nit: should this be plural like rest of them?

**Review by matejminar — APPROVED** [review]

File: static/app/types/sampling.tsx (hunk 1 above)
Comment (andriisoldatenko) [reply static/app/types/sampling.tsx:150]: it's mix `Environments` is plural; `Release` not :(

Comment (priscilawebdev) [reply static/app/types/sampling.tsx:150]: yes, thought about it too...but at least all are plural in the frontend

Comment (matejminar) [reply static/app/types/sampling.tsx:150]: I meant enum key - BOOST_KEY_TRANSACTIONS

Comment (andriisoldatenko) [reply static/app/types/sampling.tsx:150]: make sense i'm using same in py part: `BOOST_KEY_TRANSACTIONS_RULE`

Comment (priscilawebdev) [inline static/app/types/sampling.tsx:150]:
```suggestion
  BOOST_KEY_TRANSACTIONS = 'boostKeyTransaction',
```

File: static/app/views/settings/project/dynamicSampling/dynamicSampling.spec.tsx [hunk 2]
```diff
@@ -15,6 +15,7 @@ const ORG_FEATURES = [
 const dynamicSamplingBiases = [
   {id: DynamicSamplingBiasType.BOOST_LATEST_RELEASES, active: true},
   {id: DynamicSamplingBiasType.BOOST_ENVIRONMENTS, active: true},
+  {id: DynamicSamplingBiasType.BOOST_KEY_TRANSACTION, active: true},
```
Comment (priscilawebdev) [inline static/app/views/settings/project/dynamicSampling/dynamicSampling.spec.tsx:18]:
```suggestion
  {id: DynamicSamplingBiasType.BOOST_KEY_TRANSACTIONS, active: true},
```

File: static/app/views/settings/project/dynamicSampling/dynamicSampling.tsx [hunk 3]
```diff
@@ -42,6 +42,10 @@ const knowBiases = {
     label: t('Prioritize dev environments'),
     help: t('Captures more traces from environments that contain “dev” and “test”'),
   },
+  [DynamicSamplingBiasType.BOOST_KEY_TRANSACTION]: {
```
Comment (priscilawebdev) [inline static/app/views/settings/project/dynamicSampling/dynamicSampling.tsx:45]:
```suggestion
  [DynamicSamplingBiasType.BOOST_KEY_TRANSACTIONS]: {
```

File: static/app/views/settings/project/dynamicSampling/dynamicSampling.spec.tsx [hunk 4]
```diff
@@ -156,6 +171,7 @@ describe('Dynamic Sampling', function () {
           dynamicSamplingBiases: [
             {id: DynamicSamplingBiasType.BOOST_LATEST_RELEASES, active: false},
             {id: DynamicSamplingBiasType.BOOST_ENVIRONMENTS, active: true},
+            {id: DynamicSamplingBiasType.BOOST_KEY_TRANSACTION, active: true},
```
Comment (priscilawebdev) [inline static/app/views/settings/project/dynamicSampling/dynamicSampling.spec.tsx:174]:
```suggestion
            {id: DynamicSamplingBiasType.BOOST_KEY_TRANSACTIONS, active: true},
```

**Pull request merged**